src/main/java/org/folio/search/repository/classification/InstanceClassificationJdbcRepository.java

```diff
@@ -0,0 +1,164 @@
+package org.folio.search.repository.classification;
+
+import static org.folio.search.utils.JdbcUtils.getQuestionMarkPlaceholder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.PreparedStatement;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.search.model.index.InstanceSubResource;
+import org.folio.search.utils.JdbcUtils;
+import org.folio.spring.FolioExecutionContext;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Log4j2
+@Repository
+@RequiredArgsConstructor
+public class InstanceClassificationJdbcRepository implements InstanceClassificationRepository {
+
+  private static final String INSTANCE_CLASSIFICATION_TABLE_NAME = "instance_classification";
+  private static final String CLASSIFICATION_TYPE_COLUMN = "classification_type";
+  private static final String CLASSIFICATION_NUMBER_COLUMN = "classification_number";
+  private static final String TENANT_ID_COLUMN = "tenant_id";
+  private static final String INSTANCE_ID_COLUMN = "instance_id";
+  private static final String SHARED_COLUMN = "shared";
+  private static final String CLASSIFICATION_TYPE_DEFAULT = "<null>";
+
+  private static final String SELECT_ALL_SQL = "SELECT * FROM %s;";
+  private static final String SELECT_ALL_BY_INSTANCE_ID_AGG = """
+    SELECT
+        t1.classification_number,
+        t1.classification_type,
+        json_agg(json_build_object(
+            'instanceId', t1.instance_id,
+            'shared', t1.shared,
+            'tenantId', t1.tenant_id
+        )) AS instances
+    FROM %1$s t1
+    INNER JOIN %1$s t2 ON t1.classification_number = t2.classification_number
+                                        AND t1.classification_type = t2.classification_type
+                                        AND t2.instance_id IN (%2$s)
+    GROUP BY t1.classification_number, t1.classification_type;
```

Comment:
Did you check it with "EXPLAIN ANALYZE"? Will it filter by instanceId first to prevent building aggregations for all before filtering?

Reply:
And do we need filter by instanceId at all?
As I remember - we need to construct and replace full documents by type+number which will have a list of objects for each instanceId it has.
I don't see it being used for now, but I assume this is the query for constructing an elasticseach document.

Reply:
It uses 2 indexes:
![image](https://github.com/folio-org/mod-search/assets/5904625/50dd6d79-60ee-4808-a21d-15892fda60d7)
![image](https://github.com/folio-org/mod-search/assets/5904625/95f5ae6f-cebc-4f76-b2e5-2b2ad0ffc036)


As indexing of classification is part of instances indexing and preparing documents during instances processing, I think it's one of the way to find all classifications related to instances that currently in process.
This query takes all classifications that instances have and aggregate them in elastic-like document, so with this quesry we will have full document.